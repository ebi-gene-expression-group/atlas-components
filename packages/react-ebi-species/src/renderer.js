"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
