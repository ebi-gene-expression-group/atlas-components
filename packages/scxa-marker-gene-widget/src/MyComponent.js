import React from 'react'
import PropTypes from 'prop-types'

class MyComponent extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return(
      <div className={`foo`}>Bar</div>      
    )
  }
}

MyComponent.propTypes = {
  atlasUrl: PropTypes.string
}

export default MyComponent
