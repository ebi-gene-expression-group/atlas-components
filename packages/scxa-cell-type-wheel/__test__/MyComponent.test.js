// If you get ReferenceError: regeneratorRuntime is not defined use this in your tests:
// import 'core-js/stable'
// import 'regenerator-runtime/runtime'
import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

import MyComponent from '../src/MyComponent'

describe(`MyComponent`, () => {
  test(`should mount in a full DOM`, () => {
    render(<MyComponent />)
    expect(screen.getByRole(`heading`)).toBeInTheDocument()
  })

  test(`should render to static HTML`, () => {
    render(<MyComponent />)
    expect(screen.getByRole(`heading`)).toHaveTextContent(`Bar`)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<MyComponent />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
