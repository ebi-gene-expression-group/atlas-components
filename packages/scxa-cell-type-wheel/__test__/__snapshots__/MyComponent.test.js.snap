// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent matches snapshot 1`] = `
<div
  className="foo"
  role="heading"
>
  Bar
</div>
`;
